packages/core/content-manager/admin/src/pages/EditView/components/FormInputs/Relations/RelationModal.tsx

```diff
@@ -71,16 +84,38 @@ const RelationModalWrapper = ({ relation, triggerButtonLabel }: RelationModalPro
   );
   const currentDocumentMeta = useDocumentContext('RelationModalBody', (state) => state.meta);
   const changeDocument = useDocumentContext('RelationModalBody', (state) => state.changeDocument);
+  const documentHistory = useDocumentContext('RelationModalBody', (state) => state.documentHistory);
+  const setDocumentHistory = useDocumentContext(
+    'RelationModalBody',
+    (state) => state.setDocumentHistory
+  );
 
   const [isConfirmationOpen, setIsConfirmationOpen] = React.useState(false);
-  const [isNavigating, setIsNavigating] = React.useState(false);
+  const [actionPosition, setActionPosition] = React.useState<'cancel' | 'back' | 'navigate'>(
+    'cancel'
+  );
 
   const [isModalOpen, setIsModalOpen] = React.useState(false);
+  // NOTE: Not sure about this relation modal context, maybe we should move this to DocumentContext?
   // Get parent modal context if it exists
   const parentContext = useRelationModal('RelationModalWrapper', (state) => state);
-  const isNested = parentContext.depth > 0;
-  // Track this modal's depth
-  const depth = isNested ? parentContext.depth + 1 : 1;
+  // Get depth of nested modals
+  const depth = parentContext ? parentContext.depth + 1 : 0;
+  // Check if this is a nested modal
+  const isNested = depth > 0;
+
+  const addDocumentToHistory = (document: DocumentMeta) =>
+    setDocumentHistory([...documentHistory, document]);
```

Comment:
I always prefer using the callback here to avoid potential stale closure issues
```suggestion
    setDocumentHistory(prev => [...prev, document]);
```